portal-api/src/v1/users/routes.js

```diff
@@ -194,24 +212,33 @@ module.exports.resetPassword = async (req, res) => {
   });
 };
 
+/**
+ * Portal reset password route caters for following user scenarios:
+ * 1. User initiated password reset
+ * 2. Adminstrator adds a new user, where user need to specify the password.
+ */
 module.exports.resetPasswordWithToken = async (req, res, next) => {
   const { resetPwdToken } = req.params;
   const { currentPassword, password, passwordConfirm } = req.body;
 
-  if (currentPassword.trim() === '') {
-    return res.status(400).json({
-      success: false,
-      errors: {
-        count: 1,
-        errorList: {
-          currentPassword: {
-            text: 'Empty password',
+  // Only valid for user initiated password reset operation
+  if (Object.prototype.hasOwnProperty.call(req.body, 'currentPassword')) {
```

Comment:
We are already deconstructing the current Password above. Can we not validate that instead?

Reply:
I need to validate `currentPassword` against empty input but with field exists. This is because same route is used for:
1. User actioned - forgot password
2. Admin create account - user set password (where `currentPassword` field does not exists)